src/cascadia/TerminalApp/App.cpp

```diff
@@ -851,15 +843,25 @@ namespace winrt::TerminalApp::implementation
     {
         uint32_t tabIndexFromControl = 0;
         _tabView.Items().IndexOf(tabViewItem, tabIndexFromControl);
-
-        if (tabIndexFromControl == _GetFocusedTabIndex())
-        {
-            _tabView.SelectedIndex((tabIndexFromControl > 0) ? tabIndexFromControl - 1 : 1);
-        }
+        size_t focusedTabIndex = _GetFocusedTabIndex();

```

Comment:
`_GetFocusedTabIndex()` can return `-1` as a result so `size_t` is not a best option for the `focusedTabIndex` variable IMO.

Reply:
Yeah, in that - doesn't seem like the best option.
Would you have any suggestions?

Reply:
Why not just `int`?

Reply:
Visual Studio gave me warning because `_tabs.size() -1` is `size_t` and I am not sure which one to worry about more

Reply:
Why not create a (templated) utility function for vectors that returns a signed size (like proposed in c++ proposal p1227R0) and use that in these cases instead of .size() directly?

Reply:
> Why not create a (templated) utility function for vectors that returns a signed size (like proposed in c++ proposal p1227R0) and use that in these cases instead of .size() directly?

I haven't played about much with C++ so I think it is outside of my scope of ability.

Reply:
If the Microsoft devs think its a good idea i'm willing to make a follow-up PR to fix those cases.
The code is (almost) in the document though:

```
template <class C>
constexpr ptrdiff_t ssize(const C& c) noexcept
{
    return static_cast<ptrdiff_t>(std::size(c))
}
```

And then you can just use it from your code:
```
focusedTabIndex = ssize(_tabs) - 1;
```

In that case, the return value will be int and not size_t. This allows you to handle the -1 error condition properly.

Reply:
Should I maybe edit it to compare with `static_cast<ssize>(_tabs.size())` then?

Reply:
@DHowett Thoughts?

Reply:
@DHowett-MSFT

Reply:
Can this not just be `auto`?

Reply:
I'm fine with either, but it seems broadly more intuitive to use auto and let the library dictate its type. It's manifestly not a size_t, but we don't necessarily want to go through and switch it to ssize_t either.